.github/PULL_REQUEST_TEMPLATE/release-checklist.md

```diff
@@ -12,46 +12,45 @@ assignees: ''
+Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]
 
-Look for the [draft `zebrad` changelog](https://github.com/ZcashFoundation/zebra/releases) for the automatic version bump.
-This version is based on [the labels on the PRs in the release](https://github.com/ZcashFoundation/zebra/blob/main/.github/release-drafter.yml).
+The [draft `zebrad` changelog](https://github.com/ZcashFoundation/zebra/releases) will have an automatic version bump. This version is based on [the labels on the PRs in the release](https://github.com/ZcashFoundation/zebra/blob/main/.github/release-drafter.yml).
 
 Check that the automatic `zebrad` version increment is correct:
-1. If we're releasing a mainnet network upgrade, increment the `major` version of all Zebra crates
-2. If we're not releasing a mainnet network upgrade, check for features, major changes, deprecations, and removals. If this release has any, it is a `minor` release
-
-If we're not doing a `major` release, you need to check which crates have changed:
-1. Go to the zebra GitHub code page: https://github.com/ZcashFoundation/zebra
-2. Check if the last commit to each crate is a Zebra version bump. If it is a version bump, the crate has not changed since the last release.
-   
-Once you know which crates have changed:
-- [ ] Increment the crates that have new commits since the last version update
-- [ ] Increment any crates that depend on crates that have changed
-- [ ] Keep a list of the crates that haven't been incremented, to include in the PR
-
-### How to Increment Versions
 
-Zebra follows [semantic versioning](https://semver.org).
+If we're releasing a mainnet network upgrade, it is a `major` release:
+1. Increment the `major` version of _*all*_ the Zebra crates.
+2. Increment the `patch` version of the tower crates.
 
-Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]
+If we're not releasing a mainnet network upgrade, check for features, major changes, deprecations, and removals. If this release has any, it is a `minor` release:
+1. Increment the `minor` version of `zebrad`.
+2. Increment the `pre-release` version of the other crates.
+3. Increment the `patch` version of the tower crates.
 
-### Reviewing Version Bumps
+Otherwise, it is a `patch` release:
+1. Increment the `patch` version of `zebrad`.
+2. Increment the `pre-release` version of the other crates.
+3. Increment the `patch` version of the tower crates.
 
-Check for missed changes by going to:
-`https://github.com/ZcashFoundation/zebra/tree/<commit-hash>/`
-Where `<commit-hash>` is the hash of the last commit in the version bump PR.
+Zebra's Rust API is not stable or supported, so we keep all the crates on the same beta `pre-release` version.
 
-If any Zebra or Tower crates have commit messages that are **not** a version bump, we have missed an update.
-Also check for crates that depend on crates that have changed. They should get a version bump as well.
```

Comment:
Sometimes we bump crates that aren't updated, and sometimes we don't.

I think we should just update all the crates, to simplify the release process.

Reply:
> > Sometimes we bump crates that aren't updated, and sometimes we don't.
> 
> However, as I understand it, we do have a rationale for when we bump crates that aren't updated, i.e when they depend on crates that have changed... unless we haven't been doing this consistently in the past? Or am I missing something here? :)

We have been bumping all the crates for the last few releases, regardless of any changes, which is why all the Zebra and tower crates are on the same version.



> > I think we should just update all the crates, to simplify the release process. (If we want to start skipping crates again, it would really help if we automated version bumps.)
> 
> I agree that it would be good to simplify the release process and the version bumps. I think this depends on whether we want outside developers to be able to use crates individually or not though.

We decided not to guarantee any support for outside developers using the Rust APIs in the individual crates.